xcm/pallet-xcm-benchmarks/src/generic/benchmarking.rs

```diff
@@ -306,13 +305,141 @@ benchmarks! {
 		executor.holding = holding.into();
 		let instruction = Instruction::InitiateReserveWithdraw { assets: assets_filter, reserve, xcm: Xcm(vec![]) };
 		let xcm = Xcm(vec![instruction]);
-	}:{
+	}: {
 		executor.execute(xcm)?;
 	} verify {
 		// The execute completing successfully is as good as we can check.
 		// TODO: Potentially add new trait to XcmSender to detect a queued outgoing message. #4426
 	}
 
+	burn_asset {
+		let holding = T::worst_case_holding(0);
+		let assets = holding.clone();
+
+		let mut executor = new_executor::<T>(Default::default());
+		executor.holding = holding.into();
+
+		let instruction = Instruction::BurnAsset(assets.into());
+		let xcm = Xcm(vec![instruction]);
+	}: {
+		executor.execute(xcm)?;
+	} verify {
+		assert!(executor.holding.is_empty());
+	}
+
+	expect_asset {
+		let holding = T::worst_case_holding(0);
+		let assets = holding.clone();
+
+		let mut executor = new_executor::<T>(Default::default());
+		executor.holding = holding.into();
+
+		let instruction = Instruction::ExpectAsset(assets.into());
+		let xcm = Xcm(vec![instruction]);
+	}: {
+		executor.execute(xcm)?;
+	} verify {
+		// `execute` completing successfully is as good as we can check.
+	}
+
+	expect_origin {
+		let expected_origin = Parent.into();
+		let mut executor = new_executor::<T>(Default::default());
+
+		let instruction = Instruction::ExpectOrigin(Some(expected_origin));
+		let xcm = Xcm(vec![instruction]);
+		let mut _result = Ok(());
+	}: {
+		_result = executor.execute(xcm);
+	} verify {
+		assert!(matches!(_result, Err(ExecutorError {
+			xcm_error: XcmError::ExpectationFalse,
+			..
+		})));
+	}
+
+	expect_error {
```

Comment:
why is the failure case more expensive than the success?

Reply:
It actually isn't; I'm using the failure case here so that I could verify the result, because the success case looks too much like any other successes and you can't really be sure if it worked correctly.

Reply:
hmm but if the success case is is heavier, this leads to inaccurate weight (I guess this is not the case here).